File: 1-hello-world/python/producer/send.py
#!/usr/bin/env python

import pika

# Establish connection to RabbitMQ server on localhost
connection = pika.BlockingConnection()
channel = connection.channel()

# Declare queue (create if not exists)
channel.queue_declare(queue='hello')

# Send message to queue
msg = "Hello World!"
channel.basic_publish(exchange='', routing_key='hello', body=msg)
print(" [x] Sent '" + msg + "'")

# Close connection
connection.close()

File: 1-hello-world/javascript/amqplib-callback-api/producer/send.js
#!/usr/bin/env node

var amqp = require('amqplib/callback_api');

var uri = 'amqp://localhost'
var queueName = 'hello'
var msg = 'Hello World!'

// Establish connection to RabbitMQ server
amqp.connect(uri, function(err, conn) {
  conn.createChannel(function(err, ch) {
    // Declare queue (create if not exists)
    ch.assertQueue(queueName, {durable: false});
    // Send message to queue
    ch.sendToQueue(queueName, Buffer.from(msg));
    console.log(" [x] Sent '" + msg + "'");
  });
  // Close connection
  setTimeout(function() { conn.close(); process.exit(0) }, 500);
});
